modules/cluster-storage-operator.adoc

```diff
@@ -1,15 +1,49 @@
 // Module included in the following assemblies:
 //
 // *  operators/operator-reference.adoc
+// *  installing/cluster-capabilities.adoc
+
+ifeval::["{context}" == "cluster-operators-ref"]
+:operator-ref:
+endif::[]
+
+ifeval::["{context}" == "cluster-capabilities"]
+:cluster-caps:
+endif::[]
 
 [id="cluster-storage-operator_{context}"]
-= Cluster Storage Operator
+ifdef::operator-ref[= Cluster Storage Operator]
+ifdef::cluster-caps[= Cluster storage capability]
+
+ifdef::operator-ref[]
+
+[NOTE]
+====
+The Cluster Storage Operator is an optional cluster capability that can be disabled by cluster administrators during installation. For more information about optional cluster capabilities, see "Cluster capabilities" in _Installing_.
+====
+
+endif::operator-ref[]
 
 [discrete]
 == Purpose
 
+ifdef::cluster-caps[]
+
+The Cluster Storage Operator provides the features for the `Storage` capability.
+
+endif::cluster-caps[]
+
 The Cluster Storage Operator sets {product-title} cluster-wide storage defaults. It ensures a default storage class exists for {product-title} clusters.
 
+ifdef::cluster-caps[]
+[IMPORTANT]
+====
+If the cluster storage capability is disabled, the cluster will not have a default storage class. Clusters installed on {rh-virtualization-first} and {rh-openstack-first} rely on the cluster storage Operator, so keep this capability enabled on those platforms. Users with administrator privileges can create a default storage class if the cluster storage capability is disabled.
```

Comment:
As Wei pointed out, the operator also installs CSI drivers for cloud platforms supported by OCP (do we need a full list here?). When the capability is disabled, OCP does not install any CSI driver on its own, however, 3rd party CSI drivers can still be installed into the cluster, e.g. via OLM, and they will work. It is especially useful on bare metal platform, where the Cluster Storage Operator does nothing and disabling it can save some memory & CPU in constrained envs.

Reply:
I have updated the file to take this feedback into account. Please take a look

Reply:
For the update as below, if we mention manually install CSI drivers:

>  If the cluster storage capability is disabled, the cluster will not have a default `storageclass` or any CSI drivers. Users with administrator privileges can create a default `storageclass` and manually install CSI drivers if the cluster storage capability is disabled. 

We could remove this:
> Users will also need to create their own storage

Reply:
Thanks, I've updated the PR with this feedback.